app/assets/javascripts/discourse/app/components/d-lightbox.js

```diff
@@ -0,0 +1,476 @@
+  getSwipeDirection,
+  openImageInNewTab,
+  preloadItemImages,
+  scrollParentToElementCenter,
+  setCarouselScrollPosition,
+  setSiteThemeColor,
+} from "discourse/lib/lightbox/helpers";
+
+import Component from "@glimmer/component";
+import { bind } from "discourse-common/utils/decorators";
+import discourseLater from "discourse-common/lib/later";
+import { htmlSafe } from "@ember/template";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+
+export default class DLightbox extends Component {
+  @service appEvents;
+
+  @tracked callbacks = {};
+  @tracked items = [];
+  @tracked options = {};
+
+  @tracked isVisible = false;
+  @tracked isLoading = false;
+  @tracked willClose = false;
+
+  @tracked currentIndex = 0;
+  @tracked currentItem = {};
+
+  @tracked isZoomed = false;
+  @tracked isRotated = false;
+  @tracked isFullScreen = false;
+  @tracked rotationAmount = 0;
+
+  @tracked hasCarousel = false;
+  @tracked hasExpandedTitle = false;
+
+  elementId = LIGHTBOX_ELEMENT_ID;
+  titleElementId = TITLE_ELEMENT_ID;
+  animationDuration = ANIMATION_DURATION;
+
+  get layoutType() {
+    return window.innerWidth > window.innerHeight
+      ? LAYOUT_TYPES.HORIZONTAL
+      : LAYOUT_TYPES.VERTICAL;
+  }
+
+  get CSSVars() {
+    const base = "--d-lightbox-image";
+
+    const variables = [
+      `${base}-animation-duration: ${this.animationDuration}ms;`,
+    ];
+
+    if (!this.currentItem) {
+      return htmlSafe(variables.join(""));
+    }
+
+    const { width, height, aspectRatio, dominantColor, fullsizeURL, smallURL } =
+      this.currentItem;
+
+    variables.push(
+      `${base}-rotation: ${this.rotationAmount}deg`,
+      `${base}-width: ${width}px`,
+      `${base}-height: ${height}px`,
+      `${base}-aspect-ratio: ${aspectRatio}`,
+      `${base}-dominant-color: #${dominantColor}`,
+      `${base}-full-size-url: url(${encodeURI(fullsizeURL)})`,
+      `${base}-small-url: url(${encodeURI(smallURL)})`
+    );
+
+    return htmlSafe(variables.filter(Boolean).join(";"));
+  }
+
+  get HTMLClassList() {
+    const base = "d-lightbox";
+
+    const classNames = [base];
+
+    if (!this.isVisible) {
+      return classNames.join("");
+    }
+
+    classNames.push(
+      this.layoutType && `${base}--is-${this.layoutType}`,
+      this.isVisible && `${base}--is-visible`,
+      this.isLoading ? `${base}--is-loading` : `${base}--is-finished-loading`,
+      this.isFullScreen && `${base}--is-fullscreen`,
+      this.isZoomed && `${base}--is-zoomed`,
+      this.isRotated && `${base}--is-rotated`,
+      this.canZoom && `${base}--can-zoom`,
+      this.hasExpandedTitle && `${base}--has-expanded-title`,
+      this.hasCarousel && `${base}--has-carousel`,
+      this.hasLoadingError && `${base}--has-loading-error`,
+      this.willClose && `${base}--will-close`,
+      this.isRotated &&
+        this.rotationAmount &&
+        `${base}--is-rotated-${this.rotationAmount}`
+    );
```

Comment:
Can you look CSS naming is correct for you please @chapoi ?

Reply:
Okay so, the idea is that modifiers are standalone classes (not chained), with either `-` as prefix or `has-/is-`, in case of temporary/changable states.

I think based on naming this should be mostly states? So:

```suggestion
    classNames.push(
      this.layoutType && `is-${this.layoutType}`,
      this.isVisible && `is-visible`,
      this.isLoading ? `is-loading` : `is-finished-loading`,
      this.isFullScreen && `is-fullscreen`,
      this.isZoomed && `is-zoomed`,
      this.isRotated && `is-rotated`,
      this.canZoom && `can-zoom`,
      this.hasExpandedTitle && `has-expanded-title`,
      this.hasCarousel && `has-carousel`,
      this.hasLoadingError && `has-loading-error`,
      this.willClose && `will-close`,
      this.isRotated &&
        this.rotationAmount &&
        `is-rotated-${this.rotationAmount}`
    );
```